File: 0997-find-the-town-judge/0997-find-the-town-judge.js
/**
 * @param {number} n
 * @param {number[][]} trust
 * @return {number}
 */
var findJudge = function(n, trust) {
    let trusts = Array(n + 1).fill(0);
    for (let [a, b] of trust)
    {
        trusts[a] = NaN;
        ++trusts[b];
    }
    return trusts.indexOf(n - 1) || 1;
};

File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<int> trusted(n + 1);
        vector<int> trusting(n + 1);
        
        for (auto &i : trust)
        {
            ++trusted[i[0]];
            ++trusting[i[1]];
        }
        
        for (int i = 1; i <=n; ++i)
            if (trusting[i] == n - 1 && trusted[i] == 0)
                return i;
        
        return -1;
    }
};